scripts/windows/msys2_prepare.bat

```diff
@@ -1,4 +1,6 @@
 @echo off
```

Comment:
Changes in this script are basically to add more checks for env and refactor it to subroutines for easier maintenance and clarity. It does not do anything fundamentally different from the old one